File: vb.net2019/GeneralEVR/Form1.vb
Public Class Form1
    Private textFontBold As Boolean
    Private textFontItalic As Boolean
    Private textFontUnderline As Boolean
    Private textFontColor As UInt32
    Private textbgFontColor As UInt32
    Private textTransparentColor As UInt32
    Private scrollSpeed As Short
    Private alignText As Short
    Private alphaValue As Single
    Private XPos As Single
    Private YPos As Single


    Public Function Color2Uint32(ByVal clr As Color) As UInt32
        Dim t As Integer
        Dim a As Byte()
        t = ColorTranslator.ToOle(clr)
        a = BitConverter.GetBytes(t)
        Return BitConverter.ToUInt32(a, 0)
    End Function

  



    Private Sub ratioCheckBox_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) 'Handles ratioCheckBox.CheckedChanged
        Me.AxMoviePlayer1.AspectRatio = Me.ratioCheckBox.Checked
    End Sub

   


    Private Sub usePitchCheckBox_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles usePitchCheckBox.CheckedChanged
        Me.pitchGroupBox.Enabled = Me.usePitchCheckBox.Checked
    End Sub

    Private Sub playButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles playButton.Click
        Me.axMoviePlayer1.Play()
        Me.playButton.Enabled = False
        Me.stopButton.Enabled = True
        Me.pauseButton.Enabled = True

    End Sub

    Private Sub restoreTempoButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles restoreTempoButton.Click
        Me.incTempoTrackBar.Value = 0
        Me.decTempoTrackBar.Value = 0
        Me.AxMoviePlayer1.ChangeTempo(Me.incTempoTrackBar.Value)
        Me.AxMoviePlayer1.ChangeTempo(-Me.decTempoTrackBar.Value)

    End Sub

    Private Sub snapshotButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles snapshotButton.Click
        Dim strApp = Application.ExecutablePath

        Dim iFindIndex = strApp.IndexOf("WindowsApplication1")

        strApp = strApp.Substring(0, iFindIndex)

        Me.AxMoviePlayer1.SnapShot2ImageFile(strApp + "\test.bmp", 0)

        Dim fs As System.IO.FileStream
        ' Specify a valid picture file path on your computer.
        If System.IO.File.Exists(strApp + "\test.bmp") Then
            fs = New System.IO.FileStream(strApp + "\test.bmp",
            IO.FileMode.Open, IO.FileAccess.Read)
            pictureBox1.Image = System.Drawing.Image.FromStream(fs)
            fs.Close()
        End If
    End Sub

    Private Sub snapshotToHBitButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles snapshotToHBitButton.Click
        Dim hb As Integer = Me.AxMoviePlayer1.SnapShot2HBITMAP

        If hb <> 0 Then
            Dim h As New IntPtr(hb)
            Me.pictureBox1.Image = Image.FromHbitmap(h)
        End If
    End Sub

    Private Sub restorePitchButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles restorePitchButton.Click
        Me.pitchTrackBar.Value = 12
        Me.AxMoviePlayer1.ChangePitch(Me.pitchTrackBar.Value - 12)
    End Sub

   

    Private Sub stopButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles stopButton.Click
        Me.axMoviePlayer1.Stop()
        Me.stopButton.Enabled = False
        Me.pauseButton.Enabled = False
        Me.stopButton.Enabled = False
        Me.usePitchCheckBox.Enabled = True
    End Sub

    


    Private Sub drawButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs)
    End Sub

    Private Sub pauseButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles pauseButton.Click
        Me.axMoviePlayer1.Pause()
        Me.playButton.Enabled = True
        Me.pauseButton.Enabled = False
        Me.stopButton.Enabled = False
    End Sub

    

    Private Sub snapshotToPicButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles snapshotToPicButton.Click
        If (Me.pictureBox1.Image Is Nothing) Then
            Me.pictureBox1.Image = Nothing
        End If
        Try
            Me.pictureBox1.Image = Me.axMoviePlayer1.SnapShot2Picture()
        Catch ex As Exception
            Console.WriteLine(ex.ToString())
        End Try
    End Sub

    Private Sub postionButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles postionButton.Click
        Me.axMoviePlayer1.SetPos(Me.position.Value)
    End Sub

    Private Sub playbackRateButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles playbackRateButton.Click
        Try
            Me.axMoviePlayer1.PlaybackRate = Me.playbackRate.Value
        Catch ave As AccessViolationException
            MessageBox.Show("Play the file first!")
            Console.WriteLine(ave.ToString())
        Catch ex As Exception
            Console.WriteLine(ex.ToString())
        End Try
    End Sub

    Private Sub resizeButton1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles resizeButton1.Click
        Me.axMoviePlayer1.ResizeControl(100, 100)
    End Sub

    Private Sub resizeButton2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles resizeButton2.Click
        Me.axMoviePlayer1.ResizeControl(200, 200)
    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load


        Me.AxMoviePlayer1.UseEVR = True

        cbokey.Items.Add("ESC Key")
        cbokey.Items.Add("Q Key")
        cbokey.SelectedIndex = 0

        cbomousebutton.Items.Add("Left ButtonDown")
        cbomousebutton.Items.Add("Left ButtonDblClk")
        cbomousebutton.Items.Add("Right ButtonDown")
        cbomousebutton.Items.Add("Right ButtonDblClk")
        cbomousebutton.SelectedIndex = 0



        Me.pitchTrackBar.Minimum = 0
        Me.pitchTrackBar.Maximum = 24
        Me.pitchTrackBar.Value = 12

        Me.incTempoTrackBar.Maximum = 100
        Me.incTempoTrackBar.Minimum = 0

        Me.decTempoTrackBar.Maximum = 50
        Me.decTempoTrackBar.Minimum = 0

        AddHandler rightRadioButton.CheckedChanged, AddressOf rightRadioButton_CheckedChanged
        AddHandler ratioCheckBox.CheckedChanged, AddressOf ratioCheckBox_CheckedChanged
        'avoid refresh issue
        Me.AxMoviePlayer1.RegisterMsg()
    End Sub

    Private Sub Form1_FormClosing(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosingEventArgs) Handles MyBase.FormClosing
        Me.axMoviePlayer1.UnRegisterMsg()
    End Sub

    Private Sub axMoviePlayer1_OnCompleted(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles AxMoviePlayer1.OnCompleted
        Me.trackBar1.Value = Math.Min(Me.axMoviePlayer1.Duration, Me.trackBar1.Maximum)
        If (Me.loopingCheckBox.Checked) Then
            Me.axMoviePlayer1.Play()

        Else
            Me.playButton.Enabled = True
            Me.pauseButton.Enabled = False
            Me.stopButton.Enabled = False
            Me.usePitchCheckBox.Enabled = True
        End If
    End Sub

    Private Sub axMoviePlayer1_OnPlaying(ByVal sender As System.Object, ByVal e As AxMOVIEPLAYERLib._DMoviePlayerEvents_OnPlayingEvent) Handles AxMoviePlayer1.OnPlaying
        Me.trackBar1.Value = e.iCurrent
        Me.positionTextBox.Text = e.iCurrent.ToString()
    End Sub

   

  


    Private Sub rightRadioButton_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) 'Handles rightRadioButton.CheckedChanged
        If (Me.rightRadioButton.Checked) Then
            Me.AxMoviePlayer1.SetMPEG1AudioChannel(0)
        End If
    End Sub

    Private Sub leftRadioButton_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles leftRadioButton.CheckedChanged
        If (Me.leftRadioButton.Checked) Then
            Me.axMoviePlayer1.SetMPEG1AudioChannel(1)
        End If
    End Sub

    Private Sub bothRadioButton_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles bothRadioButton.CheckedChanged
        If (Me.bothRadioButton.Checked) Then
            Me.axMoviePlayer1.SetMPEG1AudioChannel(2)
        End If
    End Sub

    Private Sub soundTrackBar_Scroll(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles soundTrackBar.Scroll
        Me.axMoviePlayer1.SoundVolume = Me.soundTrackBar.Value - 10000
    End Sub

   

 

    Private Sub pitchTrackBar_Scroll(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles pitchTrackBar.Scroll
        Me.AxMoviePlayer1.ChangePitch(Me.pitchTrackBar.Value - 12)
    End Sub

   

    Private Sub decTempoTrackBar_Scroll(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles decTempoTrackBar.Scroll
        Me.axMoviePlayer1.ChangeTempo(-Me.decTempoTrackBar.Value)
    End Sub

    Private Sub incTempoTrackBar_Scroll(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles incTempoTrackBar.Scroll
        Me.axMoviePlayer1.ChangeTempo(Me.incTempoTrackBar.Value)
    End Sub

   



    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click

        If (AxMoviePlayer1.ClearFullScreenType = 0) Then

            If (cbokey.SelectedIndex = 0) Then
                AxMoviePlayer1.ClearFullScreenKey = 27
        
            ElseIf (cbokey.SelectedIndex = 1) Then
                AxMoviePlayer1.ClearFullScreenKey = 81
            End If

        Else

            AxMoviePlayer1.ClearFullScreenMouseButton = cbomousebutton.SelectedIndex


        End If


            Me.AxMoviePlayer1.ShowFullScreen(True)


    End Sub

    Private Sub trackBar1_Scroll(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles trackBar1.Scroll

    End Sub

    Private Sub groupBox5_Enter(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles groupBox5.Enter

    End Sub

    Private Sub RadioButton1_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles RadioButton1.CheckedChanged
        '  AxMoviePlayer1.ClearFullScreenType = 0
    End Sub

    Private Sub RadioButton2_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles RadioButton2.CheckedChanged
        ' AxMoviePlayer1.ClearFullScreenType = 1
    End Sub

    Private Sub RadioButton1_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles RadioButton1.Click
        AxMoviePlayer1.ClearFullScreenType = 0
    End Sub

    Private Sub RadioButton2_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles RadioButton2.Click
        AxMoviePlayer1.ClearFullScreenType = 1
    End Sub

    Private Sub fileButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles fileButton.Click
        Using ofd As New OpenFileDialog
            Dim i As Integer
            ofd.Filter = "All Files (*.*)|*.*|webm (*.webm)|*.webm|Flash (*.swf)|*.swf|AVCHD, TS(*.m2ts,ts)|*.m2ts;*.ts;*.mts;*.m2t|vob (*.vob)|*.vob|f4v (*.f4v)|*.f4v|flv (*.flv)|*.flv|mpg (*.mpg)|*.mpg|mpeg (*.mpeg)|*.mpeg|wmv (*.wmv)|*.wmv|avi (*.avi)|*.avi|mov (*.mov)|*.mov|mp4 (*.mp4)|*.mp4|3gp (*.3gp)|*.3gp"
            If (ofd.ShowDialog(Me) = DialogResult.OK) Then

                Me.AxMoviePlayer1.ScrollTextPage = 1
                Me.AxMoviePlayer1.UsePitchTempo = Me.usePitchCheckBox.Checked

                ' do not use microsoft mpeg2 decoder if you playback mpeg2 file
                Me.AxMoviePlayer1.UseDecoderFilter = True
                Me.AxMoviePlayer1.DecoderFilterType = MOVIEPLAYERLib.MYDECODERFILTERTYPE.BLACKLIST
                Me.AxMoviePlayer1.DecoderAdd2BlackList("Microsoft DTV-DVD Video Decoder")
                Me.AxMoviePlayer1.DecoderAdd2BlackList("Microsoft DTV-DVD Audio Decoder")

                Me.AxMoviePlayer1.FileName = ofd.FileName

                Me.trackBar1.Minimum = 0
                If (Me.AxMoviePlayer1.Duration > 0D) Then
                    Me.trackBar1.Maximum = Me.AxMoviePlayer1.Duration
                End If
                Me.durationTextBox.Text = Me.AxMoviePlayer1.Duration.ToString()
                Me.widthTextBox.Text = Me.AxMoviePlayer1.VideoWidth.ToString()
                Me.heightTextBox.Text = Me.AxMoviePlayer1.VideoHeight.ToString()

                Me.soundTrackBar.Value = soundTrackBar.Maximum
                AxMoviePlayer1.SoundVolume = 0

                Me.ListBoxMP4Stream.Items.Clear()

                For i = 0 To AxMoviePlayer1.AudioTrackGetCount - 1

                    ListBoxMP4Stream.Items.Add(AxMoviePlayer1.AudioTrackGetName(i))
                Next

                Me.usePitchCheckBox.Enabled = False
                Me.playButton.Enabled = True
                Me.pauseButton.Enabled = False
                Me.stopButton.Enabled = False


            End If
        End Using
    End Sub

    Private Sub ListBoxMP4Stream_Click(sender As System.Object, e As System.EventArgs) Handles ListBoxMP4Stream.Click
        Me.AxMoviePlayer1.AudioTrackSelectStream(Me.ListBoxMP4Stream.SelectedIndex)
    End Sub

    Private Sub Button3_Click(sender As System.Object, e As System.EventArgs) Handles Button3.Click
        Me.AxMoviePlayer1.StepFramePro(False)
        Me.playButton.Enabled = True
        Me.pauseButton.Enabled = False
        Me.stopButton.Enabled = False
    End Sub

    Private Sub Button2_Click(sender As System.Object, e As System.EventArgs) Handles Button2.Click
        Me.AxMoviePlayer1.StepFramePro(True)
        Me.playButton.Enabled = True
        Me.pauseButton.Enabled = False
        Me.stopButton.Enabled = False
    End Sub

    Private Sub Label2_Click(sender As System.Object, e As System.EventArgs) Handles Label2.Click

    End Sub
End Class

File: c#2010/GeneralEVR/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        private bool textFontBold;
        private bool textFontItalic;
        private bool textFontUnderline;
        private uint textFontColor;
        private uint textbgFontColor;
        private uint textTransparentColor;
        short scrollSpeed;
        short alignText;
        float alphaValue;
        float XPos;
        float YPos;

        public Form1()
        {
            InitializeComponent();
        }

        public uint Color2Uint32(Color clr)
        {

            int t;
            byte[] a;

            t = ColorTranslator.ToOle(clr);

            a = BitConverter.GetBytes(t);

            return BitConverter.ToUInt32(a, 0);

        }

       

        private void ratioCheckBox_CheckedChanged(object sender, EventArgs e)
        {
            this.axMoviePlayer1.AspectRatio = this.ratioCheckBox.Checked;
        }

       

       

        private void usePitchCheckBox_CheckedChanged(object sender, EventArgs e)
        {
            this.pitchGroupBox.Enabled = this.usePitchCheckBox.Checked;
        }

        private void playButton_Click(object sender, EventArgs e)
        {
            this.axMoviePlayer1.Play();
            this.playButton.Enabled = false;
            this.stopButton.Enabled = true;
            this.pauseButton.Enabled = true;
           
        }

        private void restoreTempoButton_Click(object sender, EventArgs e)
        {
            this.incTempoTrackBar.Value = 0;
            this.decTempoTrackBar.Value = 0;
            this.axMoviePlayer1.ChangeTempo((short)this.incTempoTrackBar.Value);
            this.axMoviePlayer1.ChangeTempo((short)(-this.decTempoTrackBar.Value));
        }

        private void snapshotButton_Click(object sender, EventArgs e)
        {
            string strApp = Application.ExecutablePath;
            int iFindIndex = strApp.IndexOf("WindowsApplication1");
            strApp = strApp.Substring(0, iFindIndex);

            this.axMoviePlayer1.SnapShot2ImageFile(strApp + "\\test.bmp", 0);

            System.IO.FileStream fs;

            if (System.IO.File.Exists(strApp + "\\test.bmp"))
            {

                fs = new System.IO.FileStream(strApp + "\\test.bmp", System.IO.FileMode.Open, System.IO.FileAccess.Read);
                this.pictureBox1.Image = System.Drawing.Image.FromStream(fs);
                fs.Close();

            }
        }

        private void snapshotToHBitButton_Click(object sender, EventArgs e)
        {
            int hb = this.axMoviePlayer1.SnapShot2HBITMAP();
            if (hb != 0)
            {
                IntPtr h = new IntPtr(hb);
                this.pictureBox1.Image = Image.FromHbitmap(h);
            }
        }

        private void restorePitchButton_Click(object sender, EventArgs e)
        {
            this.pitchTrackBar.Value = 12;
        }

        

        private void stopButton_Click(object sender, EventArgs e)
        {
            this.axMoviePlayer1.Stop();
            this.stopButton.Enabled = false;
            this.pauseButton.Enabled = false;
            this.stopButton.Enabled = false;
            this.usePitchCheckBox.Enabled = true;
        }

        

       

      
        private void pauseButton_Click(object sender, EventArgs e)
        {
            this.axMoviePlayer1.Pause();
            this.playButton.Enabled = true;
            this.pauseButton.Enabled = false;
            this.stopButton.Enabled = false;
        }

        private void fileButton_Click(object sender, EventArgs e)
        {
            using (OpenFileDialog ofd = new OpenFileDialog())
            {
                ofd.Filter = "All Files (*.*)|*.*|webm (*.webm)|*.webm|Flash (*.swf)|*.swf|AVCHD, TS(*.m2ts,ts)|*.m2ts;*.ts|Vob (*.vob)|*.vob|F4V (*.f4v)|*.f4v|FLV (*.flv)|*.flv|mpg (*.mpg)|*.mpg|mpeg (*.mpeg)|*.mpeg|wmv (*.wmv)|*.wmv|avi (*.avi)|*.avi|mov (*.mov)|*.mov|mp4 (*.mp4)|*.mp4|3gp (*.3gp)|*.3gp";
                if (ofd.ShowDialog(this) == DialogResult.OK)
                {
                  
                    this.axMoviePlayer1.ScrollTextPage = 1;
                    
                   // do not use microsoft mpeg2 decoder if you playback mpeg2 file
                    this.axMoviePlayer1.UseDecoderFilter = true;
                    this.axMoviePlayer1.DecoderFilterType = MOVIEPLAYERLib.MYDECODERFILTERTYPE.BLACKLIST;
                    this.axMoviePlayer1.DecoderAdd2BlackList("Microsoft DTV-DVD Video Decoder");
                    this.axMoviePlayer1.DecoderAdd2BlackList("Microsoft DTV-DVD Audio Decoder");

                    this.axMoviePlayer1.UsePitchTempo = this.usePitchCheckBox.Checked;
                    this.axMoviePlayer1.FileName = ofd.FileName;

                    this.trackBar1.Minimum = 0;
                    if (this.axMoviePlayer1.Duration > 0D)
                    {
                        this.trackBar1.Maximum =(int) this.axMoviePlayer1.Duration;
                    }
                    this.durationTextBox.Text = this.axMoviePlayer1.Duration.ToString();
                    this.widthTextBox.Text = this.axMoviePlayer1.VideoWidth.ToString();
                    this.heightTextBox.Text = this.axMoviePlayer1.VideoHeight.ToString();

                    this.soundTrackBar.Value = 10000;
                    axMoviePlayer1.SoundVolume = 0;

                    listBoxMP4Stream.Items.Clear();
                    for (int i = 0; i < axMoviePlayer1.AudioTrackGetCount(); i++)
                    {
                        listBoxMP4Stream.Items.Add(axMoviePlayer1.AudioTrackGetName((short)i));
                    }

                    this.usePitchCheckBox.Enabled = false;
                    this.playButton.Enabled = true;
                    this.pauseButton.Enabled = false;
                    this.stopButton.Enabled = false;

                    
                }
            }
        }

        private void snapshotToPicButton_Click(object sender, EventArgs e)
        {
            if (this.pictureBox1.Image != null)
            {
                this.pictureBox1.Image = null;
            }
            try
            {
                this.pictureBox1.Image = this.axMoviePlayer1.SnapShot2Picture();
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.ToString());
            }
            
        }

        private void postionButton_Click(object sender, EventArgs e)
        {
            this.axMoviePlayer1.SetPos((double)this.position.Value);
        }

        private void playbackRateButton_Click(object sender, EventArgs e)
        {
            try
            {
                this.axMoviePlayer1.PlaybackRate = (double)this.playbackRate.Value;
            }
            catch (AccessViolationException ave)
            {
                MessageBox.Show("Play the file first!");
                Console.WriteLine(ave.ToString());
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.ToString());
            }
        }

        private void resizeButton1_Click(object sender, EventArgs e)
        {
            this.axMoviePlayer1.ResizeControl(100, 100);
        }

        private void resizeButton2_Click(object sender, EventArgs e)
        {
            this.axMoviePlayer1.ResizeControl(200, 200);
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            this.axMoviePlayer1.UseEVR = true;

            cbokey.Items.Add("ESC Key");
            cbokey.Items.Add("Q Key");
            cbokey.SelectedIndex = 0;

            cbomousebutton.Items.Add("Left ButtonDown");
            cbomousebutton.Items.Add("Left ButtonDblClk");
            cbomousebutton.Items.Add("Right ButtonDown");
            cbomousebutton.Items.Add("Right ButtonDblClk");
            cbomousebutton.SelectedIndex = 0;

            this.textFontColor = this.Color2Uint32(Color.FromArgb(255, 0, 0));
            this.textbgFontColor = this.Color2Uint32(Color.FromArgb(0, 0, 0));
            this.textTransparentColor = this.Color2Uint32(Color.FromArgb(0, 0, 0));

            this.textFontBold = false;
            this.textFontItalic = false;
            this.textFontUnderline = false;

       

            this.pitchTrackBar.Minimum = 0;
            this.pitchTrackBar.Maximum = 24;
            this.pitchTrackBar.Value =12;

            this.incTempoTrackBar.Maximum = 100;
            this.incTempoTrackBar.Minimum = 0;

            this.decTempoTrackBar.Maximum = 50;
            this.decTempoTrackBar.Minimum = 0;

            //avoid refresh issue
            this.axMoviePlayer1.RegisterMsg();
        }

        private void Form1_FormClosing(object sender, FormClosingEventArgs e)
        {
            this.axMoviePlayer1.UnRegisterMsg();
        }

        private void axMoviePlayer1_OnCompleted(object sender, EventArgs e)
        {
            this.trackBar1.Value=Math.Min((int)this.axMoviePlayer1.Duration,this.trackBar1.Maximum);
            if(this.loopingCheckBox.Checked)
            {
                this.axMoviePlayer1.Play();
                
            }
            else
            {
                this.playButton.Enabled=true;
                this.pauseButton.Enabled=false;
                this.stopButton.Enabled=false;
                this.usePitchCheckBox.Enabled=true;
            }
        }

        private void axMoviePlayer1_OnPlaying(object sender, AxMOVIEPLAYERLib._DMoviePlayerEvents_OnPlayingEvent e)
        {
            this.trackBar1.Value = (int)e.iCurrent;
            this.positionTextBox.Text = e.iCurrent.ToString();
        }

        

    

        private void rightRadioButton_CheckedChanged(object sender, EventArgs e)
        {
            if (this.rightRadioButton.Checked)
            {
                this.axMoviePlayer1.SetMPEG1AudioChannel(0);
            }
        }

        private void leftRadioButton_CheckedChanged(object sender, EventArgs e)
        {
            if (this.leftRadioButton.Checked)
            {
                this.axMoviePlayer1.SetMPEG1AudioChannel(1);
            }
        }

        private void bothRadioButton_CheckedChanged(object sender, EventArgs e)
        {
            if (this.bothRadioButton.Checked)
            {
                this.axMoviePlayer1.SetMPEG1AudioChannel(2);
            }
        }

        private void soundTrackBar_Scroll(object sender, EventArgs e)
        {
            this.axMoviePlayer1.SoundVolume = this.soundTrackBar.Value - 10000;
        }

       

        private void pitchTrackBar_Scroll(object sender, EventArgs e)
        {
            this.axMoviePlayer1.ChangePitch((short)(this.pitchTrackBar.Value - 20));
        }

       
        private void decTempoTrackBar_Scroll(object sender, EventArgs e)
        {
            this.axMoviePlayer1.ChangeTempo((short)(-this.decTempoTrackBar.Value));
        }

        private void incTempoTrackBar_Scroll(object sender, EventArgs e)
        {
            this.axMoviePlayer1.ChangeTempo((short)this.incTempoTrackBar.Value);
        }

       

       
        private void button1_Click(object sender, EventArgs e)
        {

            if (axMoviePlayer1.ClearFullScreenType == 0)
            {
                if (cbokey.SelectedIndex == 0)
                    axMoviePlayer1.ClearFullScreenKey = 27;
             
                else if (cbokey.SelectedIndex == 1)
                    axMoviePlayer1.ClearFullScreenKey = 81;
            }
            else
            {
                axMoviePlayer1.ClearFullScreenMouseButton =(short) cbomousebutton.SelectedIndex;


            }
            axMoviePlayer1.ShowFullScreen(true);
        }

      
        private void trackBar1_Scroll(object sender, EventArgs e)
        {

        }

        private void radioButton1_CheckedChanged(object sender, EventArgs e)
        {
            axMoviePlayer1.ClearFullScreenType = 0;
        }

        private void radioButton2_CheckedChanged(object sender, EventArgs e)
        {
            axMoviePlayer1.ClearFullScreenType = 1;
        }

        private void groupBox5_Enter(object sender, EventArgs e)
        {

        }

        private void listBoxMP4Stream_Click(object sender, EventArgs e)
        {
            axMoviePlayer1.AudioTrackSelectStream((short)listBoxMP4Stream.SelectedIndex);
        }

        private void label1_Click(object sender, EventArgs e)
        {

        }

        private void button3_Click(object sender, EventArgs e)
        {
            axMoviePlayer1.StepFramePro(false);
            this.playButton.Enabled = true;
            this.pauseButton.Enabled = false;
            this.stopButton.Enabled = false;
        }

        private void button2_Click(object sender, EventArgs e)
        {
            axMoviePlayer1.StepFramePro(true);
            this.playButton.Enabled = true;
            this.pauseButton.Enabled = false;
            this.stopButton.Enabled = false;
        }


    }
}